coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -0,0 +1,938 @@
+	default:
+		return database.LogLevel(""), xerrors.Errorf("unknown log level: %d", logLevel)
+	}
+}
+
+func convertLogSource(logSource proto.LogSource) (database.LogSource, error) {
+	switch logSource {
+	case proto.LogSource_PROVISIONER_DAEMON:
+		return database.LogSourceProvisionerDaemon, nil
+	case proto.LogSource_PROVISIONER:
+		return database.LogSourceProvisioner, nil
+	default:
+		return database.LogSource(""), xerrors.Errorf("unknown log source: %d", logSource)
+	}
+}
+
+func convertComputedParameterValues(parameters []parameter.ComputedValue) ([]*sdkproto.ParameterValue, error) {
+	protoParameters := make([]*sdkproto.ParameterValue, len(parameters))
+	for i, computedParameter := range parameters {
+		converted, err := convertComputedParameterValue(computedParameter)
+		if err != nil {
+			return nil, xerrors.Errorf("convert parameter: %w", err)
+		}
+		protoParameters[i] = converted
+	}
+
+	return protoParameters, nil
+}
+
+func convertComputedParameterValue(param parameter.ComputedValue) (*sdkproto.ParameterValue, error) {
+	var scheme sdkproto.ParameterDestination_Scheme
+	switch param.DestinationScheme {
+	case database.ParameterDestinationSchemeEnvironmentVariable:
+		scheme = sdkproto.ParameterDestination_ENVIRONMENT_VARIABLE
+	case database.ParameterDestinationSchemeProvisionerVariable:
+		scheme = sdkproto.ParameterDestination_PROVISIONER_VARIABLE
+	default:
+		return nil, xerrors.Errorf("unrecognized destination scheme: %q", param.DestinationScheme)
+	}
+
+	return &sdkproto.ParameterValue{
+		DestinationScheme: scheme,
+		Name:              param.Name,
+		Value:             param.SourceValue,
+	}, nil
+}
+
+func convertWorkspaceTransition(transition database.WorkspaceTransition) (sdkproto.WorkspaceTransition, error) {
+	switch transition {
+	case database.WorkspaceTransitionStart:
+		return sdkproto.WorkspaceTransition_START, nil
+	case database.WorkspaceTransitionStop:
+		return sdkproto.WorkspaceTransition_STOP, nil
+	case database.WorkspaceTransitionDelete:
+		return sdkproto.WorkspaceTransition_DESTROY, nil
+	default:
+		return 0, xerrors.Errorf("unrecognized transition: %q", transition)
+	}
+}
+
+// WorkspaceProvisionJob is the payload for the "workspace_provision" job type.
+type WorkspaceProvisionJob struct {
+	WorkspaceBuildID uuid.UUID `json:"workspace_build_id"`
+	DryRun           bool      `json:"dry_run"`
+}
+
+// TemplateVersionDryRunJob is the payload for the "template_version_dry_run" job type.
+type TemplateVersionDryRunJob struct {
+	TemplateVersionID uuid.UUID                 `json:"template_version_id"`
+	WorkspaceName     string                    `json:"workspace_name"`
+	ParameterValues   []database.ParameterValue `json:"parameter_values"`
+}
+
+// ProvisionerJobLogsNotifyMessage is the payload published on
+// the provisioner job logs notify channel.
+type ProvisionerJobLogsNotifyMessage struct {
+	CreatedAfter int64 `json:"created_after"`
+	EndOfLogs    bool  `json:"end_of_logs,omitempty"`
+}
```

Comment:
random nit; it feels weird that these are at the bottom

Reply:
i had them at the top and it felt weird too... tbh they are weird structus